lib/json_api_resource/schemable.rb

```diff
@@ -23,6 +23,8 @@ def property(name, default = nil)
       protected
 
       def populate_missing_fields
+        self.attributes ||= HashWithIndifferentAccess.new
```

Comment:
I guess this is the core check.  It isn't really clear to me how attributes is getting nil'd, but we are just surfacing the client's attributes hash anyway, so if they nil it, we should handle that.  This is called in the constructor only though, so in there we ought to have ensured we have attributes.  The other changes here are trying to make that happen (but it seemed like it already was?)